crates/driver/src/domain/competition/score.rs

```diff
@@ -33,16 +36,68 @@ impl From<eth::U256> for Score {
     }
 }
 
+impl std::ops::Add for Score {
+    type Output = Self;
+
+    fn add(self, other: Self) -> Self {
+        Self(self.0 + other.0)
+    }
+}
+
+impl num::Zero for Score {
+    fn zero() -> Self {
+        Self(eth::U256::zero())
+    }
+
+    fn is_zero(&self) -> bool {
+        self.0.is_zero()
+    }
+}
+
 /// Represents the probability that a solution will be successfully settled.
-#[derive(Debug, Clone)]
+#[derive(Debug, Copy, Clone)]
 pub struct SuccessProbability(pub f64);
 
+impl From<f64> for SuccessProbability {
+    fn from(value: f64) -> Self {
+        Self(value)
+    }
+}
+
+/// Represents the objective value of a solution. This is not an artifical value
+/// like score. This is a real value that solution provides and it's based on
+/// the surplus and fees of the solution.
+#[derive(Debug, Default, PartialEq, Eq, PartialOrd, Ord, Copy, Clone)]
+pub struct ObjectiveValue(pub eth::U256);
+
+impl From<eth::U256> for ObjectiveValue {
+    fn from(value: eth::U256) -> Self {
+        Self(value)
+    }
+}
+
+impl std::cmp::PartialEq<ObjectiveValue> for Score {
```

Comment:
Maybe a small comment why comparing `ObjectiveValue` with `Score` is reasonable would be cool.